### ISSUE TITLE
In_tail doesnt work as expected and causes dropped logs

### ISSUE THREAD

**Issue by microsoft-saya** [fluent/fluentd#4673]
### Describe the bug

We are using fluentd with in_tail to foward our logs to other services. However, we are seeing an issue where after a log rotation, in_tail stops working and causes missing logs. 

I see the following in the current reading position when the issue occurs:
`/var/log/containers/file.log     ffffffffffffffff        00000000001bba10`

### To Reproduce

Doesn't happen always. Just randomly.

### Expected behavior

After log rotation, the tail functionality should work properly and update the correct reading position which will result in no loss of logs.

### Your Environment

```markdown
- Fluentd version: 1.15.3
- Package version: td-agent version: 4.4.2
- Operating system: Ubuntu
- Kernel version:
```

### Your Configuration

```apache
<source>
  @type tail
  path /var/log/containers/*.log
  exclude_path ["/var/log/containers/geneva-services*.log", "/var/log/containers/prom-mdm-converter*.log"]
  pos_file /var/log/fluentd-containers.log.pos
  tag kubernetes.*
  read_from_head true
  follow_inodes true
  <parse>
    # Reads logs in CRI format for Kubernetes v1.19+
    # The CRI format is documented here: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/kubelet-cri-logging.md
    @type regexp
    expression /^(?<time>.+) (?<stream>stdout|stderr)( (?<logtag>.))? (?<log>.*)$/
    time_format '%Y-%m-%dT%H:%M:%S.%N%:z'
    keep_time_key true
  </parse>
</source>
```

### Your Error Log

```shell
In fluentd-containers.log.pos file:
/var/log/containers/file.log     ffffffffffffffff        00000000001bba10

Fluentd logs:
2024-10-14T19:33:21.803662079Z stdout F 2024-10-14 19:33:21 +0000 [info]: #0 following tail of /var/log/containers/file.log
2024-10-14T19:33:22.256006681Z stdout F 2024-10-14 19:33:22 +0000 [info]: #0 detected rotation of /var/log/containers/file.log; waiting 5 seconds
2024-10-14T19:33:25.844191091Z stdout F 2024-10-14 19:33:25 +0000 [info]: Exception encountered parsing namespace watch event. The connection might have been closed. Sleeping for 1 seconds and resetting the namespace watcher.error reading from socket: Connection reset by peer
```

**Comment by ashie** [comment]
It seems a known bug and we believe it's already fixed in v1.16.3 or later:
https://github.com/fluent/fluentd/releases/tag/v1.16.2
https://github.com/fluent/fluentd/releases/tag/v1.16.3

Could you try latest version?

**Comment by ashie** [comment]
So I close this for now.
Please let us know if you have similar issue with latest version.
